File: app/src/main/java/com/example/helloandriod/SecondActivity.java
package com.example.helloandriod;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class SecondActivity extends AppCompatActivity {
    private EditText enteredName;
    private Button submitButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        enteredName = findViewById(R.id.enteredName);
        submitButton = findViewById(R.id.submitButton);

        String greetingMessage = getIntent().getStringExtra("greeting");

        submitButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = enteredName.getText().toString();
                Intent intent = new Intent();
                intent.putExtra("name", name);
                setResult(RESULT_OK, intent);
                finish();
            }
        });
    }
}

File: app/src/main/java/com/example/helloandriod/SecondActivity.kt
package com.example.helloandriod

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

class SecondActivity : AppCompatActivity() {
    private lateinit var enteredName: EditText
    private lateinit var submitButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        enteredName = findViewById(R.id.enteredName)
        submitButton = findViewById(R.id.submitButton)

        submitButton.setOnClickListener {
            val name = enteredName.text.toString()
            val intent = Intent().apply {
                putExtra("name", name)
            }
            setResult(RESULT_OK, intent)
            finish()
        }
    }
}